GDJS/GDJS/Extensions/Builtin/CommonInstructionsExtension.cpp

```diff
@@ -465,27 +465,41 @@ CommonInstructionsExtension::CommonInstructionsExtension() {
             "const $VALUE_ITERATOR_REFERENCE = runtimeScene.getVariables().get($VALUE_ITERATOR_VARIABLE_NAME);\n";
         outputCode +=
             "const $ITERABLE_REFERENCE = runtimeScene.getVariables().get($ITERABLE_VARIABLE_NAME);\n";
+
+        // Do not execute the loop on non iterables
+        outputCode += "if(!$ITERABLE_REFERENCE.isPrimitive()) {\n";
         
         // Begin the for loop
         outputCode +=
             "for(\n"
             "    const $ITERATOR_KEY in \n"
-            "    $ITERABLE_REFERENCE.getAllChildren()\n"
+            "    $ITERABLE_REFERENCE.getType() === \"structure\"\n"
+            "      ? $ITERABLE_REFERENCE.getAllChildren()\n"
+            "      : $ITERABLE_REFERENCE.getType() === \"array\"\n"
+            "        ? $ITERABLE_REFERENCE.getAllChildrenArray()\n"
+            "        : []\n"
             ") {\n";
         
         // If variables are defined, store the value in them
         if(keyIteratorExists) outputCode += 
-            "    $KEY_ITERATOR_REFERENCE.setString($ITERATOR_KEY);\n";
+            "    $ITERABLE_REFERENCE.getType() === \"structure\"\n"
```

Comment:
Nitpicking: this could be a if that is generated right? No need for a ternary, or even badVariable here?